### ISSUE TITLE
Bug: CI Pipeline is not publishing macOS binaries

### ISSUE THREAD

**Issue by sambacha** [foundry-rs/foundry#5154]
### Component

Forge, Cast, Anvil, Chisel, Other (please describe)

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### Operating System

macOS (Apple Silicon)

## macOS Binaries missing from GitHub Release Artifacts 

see latest release https://github.com/foundry-rs/foundry/releases/tag/nightly-2ffa619a1cda6b39e57d5e0ee567ffc1e3cc42d9

previous release: https://github.com/foundry-rs/foundry/releases/tag/nightly-03aa926203e6dabdcf9b7e59219271294717c2eb

It only goes back to the second latest release where they are missing. 

### GraphQL Query

```graphql
{
  repository(owner: "foundry-rs", name: "foundry") {
    releases(first: 100) {
      nodes {
        tag {
          name
        }
        releaseAssets(first: 100) {
          nodes {
            name
          }
        }
      }
    }
  }
}
```

#### Response 

```jsonc
{
  "data": {
    "repository": {
      "releases": {
        "nodes": [
          {
            "tag": {
              "name": "nightly-2ffa619a1cda6b39e57d5e0ee567ffc1e3cc42d9"
            },
            "releaseAssets": {
              "nodes": [
                {
                  "name": "foundry_nightly_linux_amd64.tar.gz"
                },
                {
                  "name": "foundry_man_nightly.tar.gz"
                },
                {
                  "name": "foundry_nightly_linux_arm64.tar.gz"
                }
              ]
            }
          },
          {
            "tag": {
              "name": "nightly-03aa926203e6dabdcf9b7e59219271294717c2eb"
            },
            "releaseAssets": {
              "nodes": [
                {
                  "name": "foundry_man_nightly.tar.gz"
                },
                {
                  "name": "foundry_nightly_linux_amd64.tar.gz"
                }
              ]
            }
          },
          {
            "tag": {
              "name": "nightly-08a629a5e3f33b6750222526c8f58e2beb83f794"
            },
            "releaseAssets": {
              "nodes": [
                {
                  "name": "foundry_nightly_linux_amd64.tar.gz"
                },
                {
                  "name": "foundry_man_nightly.tar.gz"
                },
                {
                  "name": "foundry_nightly_linux_arm64.tar.gz"
                },
                {
                  "name": "foundry_nightly_win32_amd64.zip"
                },
                {
                  "name": "foundry_nightly_darwin_amd64.tar.gz"
                },
                {
                  "name": "foundry_nightly_darwin_arm64.tar.gz"
                }
              ]
            }
          }
}
}
}
```

**Comment by sambacha** [comment]
I think this may have been an issue with compiling anvil, looking at my CI logs:

https://github.com/sambacha/foundry/actions/runs/5171908900/jobs/9315744971#step:7:34

> CI Timed out at 5h 48m 50s runtime

```console
Run actions-rs/cargo@v[1](https://github.com/sambacha/foundry/actions/runs/5171908900/jobs/9315744971#step:7:1)
/Users/runner/.cargo/bin/cargo build --release --bins --target aarch64-apple-darwin
   Compiling foundry-config v0.2.0 (/Users/runner/work/foundry/foundry/config)
   Compiling foundry-macros-impl v0.0.0 (/Users/runner/work/foundry/foundry/macros/impl)
   Compiling foundry-abi v0.1.0 (/Users/runner/work/foundry/foundry/abi)
   Compiling foundry-macros v0.1.0 (/Users/runner/work/foundry/foundry/macros)
   Compiling foundry-cli v0.2.0 (/Users/runner/work/foundry/foundry/cli)
   Compiling chisel v0.1.1 (/Users/runner/work/foundry/foundry/chisel)
   Compiling anvil v0.1.0 (/Users/runner/work/foundry/foundry/anvil)
   Compiling anvil-rpc v0.1.0 (/Users/runner/work/foundry/foundry/anvil/rpc)
   Compiling anvil-server v0.1.0 (/Users/runner/work/foundry/foundry/anvil/server)
   Compiling foundry-common v0.1.0 (/Users/runner/work/foundry/foundry/common)
   Compiling forge-fmt v0.2.0 (/Users/runner/work/foundry/foundry/fmt)
   Compiling foundry-utils v0.2.0 (/Users/runner/work/foundry/foundry/utils)
   Compiling forge-doc v0.1.0 (/Users/runner/work/foundry/foundry/doc)
   Compiling foundry-evm v0.2.0 (/Users/runner/work/foundry/foundry/evm)
   Compiling forge v0.2.0 (/Users/runner/work/foundry/foundry/forge)
   Compiling cast v0.2.0 (/Users/runner/work/foundry/foundry/cast)
   Compiling ui v0.2.0 (/Users/runner/work/foundry/foundry/ui)
   Compiling anvil-core v0.1.0 (/Users/runner/work/foundry/foundry/anvil/core)
Error: The operation was canceled.
```


## Draft Fix

Something like this

```yaml
  completion:
    name: "CI Pass Check"
    needs: [release, cleanup] # Maybe different job than cleanup
    if: ${{ success() || failure() }}
    runs-on: "ubuntu-22.04"
    steps:
      - name: CI Passed
        if: ${{ (needs.supported.result == 'success' && needs.other-check.result == 'success') || (needs.supported.result == 'skipped' && needs.other-check.result == 'skipped') }}
        run: |
          echo "CI workflow completed successfully.";

      - name: CI Failed
        if: ${{ needs.supported.result == 'failure' && needs.other-check.result == 'failure' }}
        run: |
          echo "CI workflow failed at some point.";
          exit 1;

```

**Comment by Evalir** [comment]
ty sir for this—looked into it, turns out this was caused by IPC not compiling on windows. This was fixed today: 087837ec9748a21b8732a2085521194e1f1346c9

a new job is running which should pass https://github.com/foundry-rs/foundry/actions/runs/5261179484
